GVFS/GVFS.Common/ProductUpgrader.Shared.cs

```diff
@@ -14,17 +14,27 @@ public partial class ProductUpgrader
         private const string GVFSInstallerFileNamePrefix = "SetupGVFS";
         private const string VFSForGitInstallerFileNamePrefix = "VFSForGit";
 
-        public static bool IsLocalUpgradeAvailable()
+        public static bool IsLocalUpgradeAvailable(string[] installerExtensions)
         {
             string downloadDirectory = GetAssetDownloadsPath();
             if (Directory.Exists(downloadDirectory))
             {
-                const string PotentialInstallerName = "*VFS*.*";
-                string[] installers = Directory.GetFiles(
-                    downloadDirectory,
-                    PotentialInstallerName, 
-                    SearchOption.TopDirectoryOnly);
-                return installers.Length > 0;
+                HashSet<string> extensions = new HashSet<string>(installerExtensions, StringComparer.OrdinalIgnoreCase);

```

Comment:
Both of the platform implementations currently have a single extension, so do we need to construct a new HashSet for this every time we run the hooks?

Reply:
Right. Have updated both platforms to return ```string``` rather than an ```array``` for ```InstallerExtension```. Replaced HashSet with string.